REVIEW_GUIDELINES.md

```diff
@@ -0,0 +1,63 @@
+Every ROS distribution is on a different sync schedule.
+
+Types of Pull Requests
+----------------------
+There are generally 5 different types of pull requests that are opened against this repository (these are in order of most likely to least likely to happen):
+
+1.  A version update to an existing package in a ROS distribution.  An example of this kind of PR is [25822](https://github.com/ros/rosdistro/pull/25822).  In these types of pull requests, the package owner is updating a major, minor, patch, or debinc number in the package.  As long as the ROS distribution isn’t in a “sync freeze”, these PRs will be merged without further review.  If the ROS distribution is in a "sync freeze", the PR will be approved with a note saying something like “Holding for sync freeze” with a link to the relevant discourse post. Once the distribution is out of sync freeze, these PRs will then be merged.
+
+1.  A new binary package in a ROS distribution.  An example of this kind of PR is [25857](https://github.com/ros/rosdistro/pull/25857).  An entry is considered a new binary package if it has a `release` field.  These PRs should generally have been opened via bloom; if they aren’t, the reviewer will ask why the submitter hasn't used bloom.  There are two sub-categories of this kind of PR:
+    1.  If the package has been released into a prior ROS distribution, then the reviewer will ensure that the URLs to the doc, release, and source entries are the same as in the previous releases (this will be checked either by looking for the package in a previous distros distribution.yaml file, or by looking at the release repository for a branch named release/{distro}).  If they are the same, and the ROS distribution isn’t in a "sync freeze", then the PR will be merged without further review.  If the URLs are different, then the reviewer will ask the submitter why the URLs are different and whether it can be combined into the original one.
+    1.  For packages that are brand-new to the ROS ecosystem, some due diligence will be done on the source and release repositories.  The source repository should have a top-level LICENSE file, and the license should be one of the open-source ones listed in [https://opensource.org/licenses](https://opensource.org/licenses) (preferably one of the "Popular licenses").  That license should also be reflected in the package.xml file of all sub-packages in the repository.  Both the source and release repositories should be publicly accessible.
+
+1.  A new source or documentation package in a ROS distribution.  An example of this kind of PR is [26383](https://github.com/ros/rosdistro/pull/26383).  An entry is considered a source or documentation package if it has no `release` field, but has a `source` or `doc` field (or both).  These PRs need not have been opened with bloom.  There are two sub-categories of this kind of PR:
```

Comment:
are only repositories that have ROS package(s) accepted here? (i.e. packages with a `package.xml` manifest)

Reply:
> are only repositories that have ROS package(s) accepted here? (i.e. packages with a `package.xml` manifest)

It's complicated.  For *most* user packages, yes, they need to have a package.xml.  We make certain exceptions for vendored libraries, or ones that really aren't ROS-native.  For instance, the source of https://github.com/eProsima/Fast-CDR is not a ROS package, which makes sense; it isn't specific to ROS.  However, in order to have our buildfarm work with it, it does need to have a package.xml.  To resolve that conflict, we patch in a package.xml in the release repository: https://github.com/ros2-gbp/fastcdr-release/blob/ae9128532d616502ecaf603de254a275a31cac43/foxy/package.xml .

I would say that the general rule is that packages should be ROS packages and have a package.xml.  If the submitter really knows what they are doing, and understands about patching in during release time, then we can consider taking non-ROS packages in.

Reply:
I've cleaned it up a bit in 3991c32.  What do you think?